### PR TITLE
Rework style (again)

### PR DESCRIPTION
**Pull request by fenjalien**
- Reworks styling to work in a kind of simpler way.
- Fixes wrong styling in an example in the manual
- Added detailed description of how styling works

### CODE REVIEW

File: src/draw/shapes.typ [hunk 1]
```diff
@@ -665,17 +666,21 @@
       (calc.max(from.at(0), to.at(0)), calc.max(from.at(1), to.at(1)))
     )
 
-    let style = styles.resolve(ctx.style, style)
-    if help-lines {
+    let style = styles.resolve(ctx.style, merge: style, root: "grid", base: (
```
Comment (johannes-wolf) [inline src/draw/shapes.typ:669]: Any reason not having this in the default style dictionary?

File: src/lib/axes.typ [hunk 2]
```diff
@@ -30,6 +30,18 @@
     stroke: (paint: gray, dash: "dotted"),
     fill: none
   ),
+  x: (
```
Comment (johannes-wolf) [inline src/lib/axes.typ:33]: If we add `x` and `y`, we should also add `x2` and `y2`.

File: src/styles.typ [hunk 3]
```diff
@@ -108,68 +124,96 @@
...
-            v = parent.at(k)
-            if v != auto {
-              dict.insert(k, v)
-              break
+      let is-dict = type(v) == dictionary
+      if is-dict or v == auto {
+        for ancestor in ancestors {
+          if k in ancestor {
+            // If v is auto and the ancestor's value is not auto, update v.
+            if (ancestor.at(k) != auto and v == auto) {
```
Comment (johannes-wolf) [inline src/styles.typ:181]: Nitpick: remove parentheses.

**Review by johannes-wolf — CHANGES_REQUESTED** [review]
Some small things only.
The changed `arc` test image does not match the output. Current reference images are still Typst 0.9.0.

File: src/draw/shapes.typ (hunk 1 above)
Comment (fenjalien) [reply src/draw/shapes.typ:669]: It doesn't really matter where the default of an element's style is. Really `styles.default` can just have stroke, fill, radius, padding, shorten and mark keys in it. Then use `base` for each element and it'll work just the same...

File: tests/axes/test.typ [hunk 4]
```diff
@@ -1,14 +1,19 @@
...
-  set-style(axes: (padding: .75))
-  set-style(axes: (x: (stroke: red)))
-  set-style(axes: (y: (stroke: green, tick: (stroke: blue, length: .3))))
+  set-style(axes: (
+    stroke: blue,
+    padding: .75,
+    x: (stroke: red),
+    y: (stroke: green, tick: (stroke: blue, length: .3))
+  ))
+  // set-style(axes: (padding: .75))
```
[the remark this answers is not in the document]
Comment (fenjalien) [reply tests/axes/test.typ]: removed

File: src/styles.typ [hunk 5]
```diff
@@ -26,78 +26,94 @@
...
+  circle: (
+    radius: auto,
+    stroke: auto,
+    fill: auto
+  ),
+  rect: (
+    stroke: auto,
+    fill: auto,
   ),
-  mark: _default-mark,
```
[the remark this answers is not in the document]
Comment (fenjalien) [reply src/styles.typ:34]: Moved comments and removed

File: src/lib/axes.typ (hunk 2 above)
Comment (fenjalien) [reply src/lib/axes.typ:33]: There is no x2 and y2 but top, right, bottom and left. Added the defaults within the first axis-settings loop.

File: src/styles.typ (hunk 3 above)
Comment (fenjalien) [reply src/styles.typ]: Removed

**Pull request merged**